internal/controllers/machine/machine_controller_status_test.go

```diff
@@ -0,0 +1,1533 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "machine-test",
+			Namespace: metav1.NamespaceDefault,
+		},
+		Spec: clusterv1.MachineSpec{
+			InfrastructureRef: corev1.ObjectReference{
+				APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+				Kind:       "GenericInfrastructureMachine",
+				Name:       "infra-machine1",
+			},
+		},
+	}
+
+	testCases := []struct {
+		name             string
+		machine          *clusterv1.Machine
+		node             *corev1.Node
+		expectConditions []metav1.Condition
+	}{
+		{
+			name:    "get NodeHealthy and NodeReady from node",
+			machine: defaultMachine.DeepCopy(),
+			node: &corev1.Node{
+				Status: corev1.NodeStatus{
+					Conditions: []corev1.NodeCondition{
+						{Type: corev1.NodeReady, Status: corev1.ConditionFalse, Reason: "SomeReason", Message: "Some message"},
+						{Type: corev1.NodeMemoryPressure, Status: corev1.ConditionFalse},
+						{Type: corev1.NodeDiskPressure, Status: corev1.ConditionFalse},
+						{Type: corev1.NodePIDPressure, Status: corev1.ConditionFalse},
+					},
+				},
+			},
+			expectConditions: []metav1.Condition{
+				{
+					Type:    clusterv1.MachineNodeHealthyV1Beta2Condition,
+					Status:  metav1.ConditionFalse,
+					Reason:  "SomeReason",
+					Message: "Node Ready: condition is False",
+				},
+				{
+					Type:    clusterv1.MachineNodeReadyV1Beta2Condition,
+					Status:  metav1.ConditionFalse,
+					Reason:  "SomeReason",
+					Message: "Some message (from Node)",
+				},
+			},
+		},
+		{
+			// TODO: handle missing conditions in summarize node conditions.
+			name:    "NodeReady missing from node",
+			machine: defaultMachine.DeepCopy(),
+			node: &corev1.Node{
+				Status: corev1.NodeStatus{
+					Conditions: []corev1.NodeCondition{
+						{Type: corev1.NodeMemoryPressure, Status: corev1.ConditionFalse},
+						{Type: corev1.NodeDiskPressure, Status: corev1.ConditionFalse},
+						{Type: corev1.NodePIDPressure, Status: corev1.ConditionFalse},
+					},
+				},
+			},
+			expectConditions: []metav1.Condition{
+				{
+					Type:    clusterv1.MachineNodeHealthyV1Beta2Condition,
+					Status:  metav1.ConditionUnknown,
+					Reason:  clusterv1.MachineNodeConditionNotYetReportedV1Beta2Reason,
+					Message: "Node Ready: condition not yet reported",
+				},
+				{
+					Type:   clusterv1.MachineNodeReadyV1Beta2Condition,
+					Status: metav1.ConditionUnknown,
+					Reason: clusterv1.MachineNodeConditionNotYetReportedV1Beta2Reason,
+				},
+			},
+		},
+		{
+			name: "node not found while machine is deleting",
+			machine: func() *clusterv1.Machine {
+				m := defaultMachine.DeepCopy()
+				m.SetDeletionTimestamp(&metav1.Time{Time: time.Now()})
+				return m
+			}(),
+			node: nil,
+			expectConditions: []metav1.Condition{
+				{
+					Type:    clusterv1.MachineNodeHealthyV1Beta2Condition,
+					Status:  metav1.ConditionUnknown,
+					Reason:  clusterv1.MachineNodeDeletedV1Beta2Reason,
+					Message: "Node has been deleted",
```

Comment:
Technially we can only say that if we know that the Node ever existed (via nodeRef?)

Reply:
Fixed